File: test/anagram-service.test.js
const AnagramService = require('../src/anagram-service');

test('should load a dictionary and find anagrams', async () => {
  let service = new AnagramService('test/test-list.txt');
  await service.loadDictionary();

  let results = await service.getAnagrams('friend');

  const expected = ['finder', 'friend', 'redfin', 'refind'];
  expect(results).toEqual(expect.arrayContaining(expected));
});

File: src/test/com/move/exercise/AnagramServiceTest.java
package com.move.exercise;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public class AnagramServiceTest {
	
	@Test
	public void testService() {
		String[] expected = {"finder", "friend", "redfin", "refind"};

		AnagramService service = new AnagramService();
		String[] result = service.findAnagrams("friend");
		
		assertEquals(expected, result);
	}
}
